src/SonarScanner.MSBuild.PreProcessor/HostInfo.cs

```diff
@@ -85,7 +85,9 @@ public record CloudHostInfo(string ServerUrl, string ApiBaseUrl, string Region)
     public static new CloudHostInfo FromProperties(ILogger logger, string sonarHostUrl, string sonarCloudUrl, string apiBaseUrl, string region)
     {
         region = region?.Trim().ToLower() ?? string.Empty;
-        if (KnownRegions.SingleOrDefault(x => x.Region == region) is { } knownRegion)
+        var regionFromUrl = region == string.Empty ? KnownRegions.FirstOrDefault(x => UrlEquals(x.ServerUrl, sonarHostUrl) || UrlEquals(x.ServerUrl, sonarCloudUrl)) : null;
+        var knownRegion = regionFromUrl ?? KnownRegions.SingleOrDefault(x => x.Region == region);
```

Comment:
The URL takes precedence, because when someone does this mistake
```
Scanner.MsBuild.exe /d:sonar.host.url=https://sonarcloud.io /d:sonar.region=us
```
we infer EU URL with EU API URL and it has a chance to work. While if we'd assume US region, EU URL would win later and be paired with assumed US API URL and that would not work.

Is that correct?

Reply:
We would assume us region and have a mixture of EU server url and us api url.
See also my comments on the test case. Let's take a decision at that discussion.
https://github.com/SonarSource/sonar-scanner-msbuild/pull/2409#discussion_r2012181018